components/clp-package-utils/clp_package_utils/controller.py

```
@@ -0,0 +1,773 @@
+        return env_vars
+
+    def _read_and_update_settings_json(
+        self, settings_file_path: pathlib.Path, updates: dict[str, Any]
+    ) -> dict[str, Any]:
+        """
+        Reads and updates a settings JSON file.
+
+        :param settings_file_path:
+        :param updates:
+        """
+        with open(settings_file_path, "r") as settings_json_file:
```

Ruff (0.14.0)

Unnecessary mode argument

Remove mode argument

(UP015)